script.js

```diff
@@ -73,159 +73,201 @@ if (
     };
   });
 
-  //------ Rating filter -------
+  //--------- FILTER BY RATING ----------
   const ratingInput = document.getElementsByName("rating");
 
-  function ratingFilter() {
-    let cards = document.querySelectorAll(".challenge-item");
-    for (let i = 0; i < cards.length; i++) {
-      if (
-        cards[i].querySelector("ul.rating").ariaValueNow >=
-          current_star_level_from &&
-        cards[i].querySelector("ul.rating").ariaValueNow <=
-          current_star_level_to
-      ) {
-        cards[i].classList.remove("is-hidden");
-      } else {
-        cards[i].classList.add("is-hidden");
+  class RatingFilter {
+    constructor(list) {
+    }
+    challengeMatch(card) {
+      let cards = document.querySelectorAll(".challenge-item");
+      for (let i = 0; i < cards.length; i++) {
+        if (
+          card.querySelector("ul.rating").ariaValueNow >=
+            current_star_level_from &&
+          card.querySelector("ul.rating").ariaValueNow <=
+            current_star_level_to
+        ) {
+          return true;
+        } else {
+          return false;
+        }
       }
+      
     }
   }
 
-  //Trigger for Rating Filter
+  //----------- FILTER BY RATING TRIGGER ---------
   for (let rating of ratingInput) {
     rating.addEventListener("click", () => {
-      ratingFilter();
+      this.filter = new FilterCollection(this);
```

Comment:
This for loop (like all of the other "trigger" sections, for every filter) are executed in the global scope. They are not written inside a function, or inside a method in a class, but inside an if statement in the global scope of this javascript file.

This means that the `this` keyword does not refer to any class instance, but to the global object, which is `window`.

The `this` keyword, and the use of classes, makes this code look object-oriented, but it really isn't object oriented. Instead, it uses classes as if they were regular functions, and it uses `this` in a way that is almost no different from omitting `this.` entirely.